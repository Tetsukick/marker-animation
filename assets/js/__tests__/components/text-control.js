/* eslint-disable no-magic-numbers */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MyTextControl from '../../src/gutenberg/components/text-control';

describe('MyTextControl', () => {
	it('should render text control', () => {
		const onChange = jest.fn(value => {
			expect(value).toBe('xyz');
		});
		const args     = {
			value: {
				start: 0,
				end: 1,
				text: 'test',
			},
		};
		const wrapper  = mount(<MyTextControl
			id='test-id'
			key='test-key'
			label='test-label'
			args={args}
			value='abc'
			onChange={onChange}
		/>);

		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('render1');
		expect(wrapper.find('#test-id').hostNodes()).toHaveLength(1);

		wrapper.find('#test-id').hostNodes().simulate('change', { target: { value: 'xyz' } });
		expect(onChange).not.toBeCalled();
		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('changed1');

		wrapper.find('#test-id').hostNodes().simulate('blur');
		expect(onChange).toBeCalled();
		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('blur1');
	});

	it('should render text control', () => {
		const onChange = jest.fn();
		const args     = {
			value: {
				start: 0,
				end: 1,
				text: 'test',
			},
		};
		const wrapper  = mount(<MyTextControl
			id='test-id'
			key='test-key'
			label='test-label'
			args={args}
			value='abc'
			onChange={onChange}
		/>);

		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('render2');
		expect(wrapper.find('#test-id').hostNodes()).toHaveLength(1);

		wrapper.find('#test-id').hostNodes().simulate('blur');
		expect(onChange).not.toBeCalled();
		expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot('blur2');
	});
});
