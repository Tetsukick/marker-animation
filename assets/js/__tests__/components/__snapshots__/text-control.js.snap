// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTextControl should render text control: blur1 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-0"
    >
      test-label
    </label>
    <input
      className="components-text-control__input"
      id="test-id"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="xyz"
    />
  </div>
</div>
`;

exports[`MyTextControl should render text control: blur2 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-1"
    >
      test-label
    </label>
    <input
      className="components-text-control__input"
      id="test-id"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="abc"
    />
  </div>
</div>
`;

exports[`MyTextControl should render text control: changed1 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-0"
    >
      test-label
    </label>
    <input
      className="components-text-control__input"
      id="test-id"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="xyz"
    />
  </div>
</div>
`;

exports[`MyTextControl should render text control: render1 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-0"
    >
      test-label
    </label>
    <input
      className="components-text-control__input"
      id="test-id"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="abc"
    />
  </div>
</div>
`;

exports[`MyTextControl should render text control: render2 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-1"
    >
      test-label
    </label>
    <input
      className="components-text-control__input"
      id="test-id"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="abc"
    />
  </div>
</div>
`;
